File: JavaKotlinDers15Navigation/app/src/main/java/com/example/javakotlinders15navigation/MainActivity.java
package com.example.javakotlinders15navigation;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

/**
 * developer.android.com adresine gidiyoruz.
  Jetpack'e giriyoruz. Jetpack: Android için geliştirilen bir kütüphane bütünü
 * navigation: görünümler arası gezmeyi kolaylaştıran bir kütüphane. Özellikle fragment
  kullanırken navigation da kullanmak iyi olur.
 * gradle:app'e ve gradle:project'e eklemem gerekenleri ekledim. Sitede de yazıyor zaten
 *
 * navigasyon grafiği oluşturmak için: res- sağ tık- new -android resource file diyoruz.
 resource type'ını navigation seçmemiz lazım oluştururken. !!
 *navigasyonumu oluşturduk iki fragment arasındaki sonra maine geliyoruz
 xml_mainde tek bir görünüm olacak o da navhost !!! onu xmle atıyoruz ve bitiyor.
 *şimdi kodlarını yazmamız gerekiyor. Build-Rebuild Project diyoruz bizim için hazır sınıf oluşturuyor.
 directions sınıfları falan oluştu. First Fragmente gidelim butona basılında ne olacağını yazdık
 *Şimdi veri aktaracağız bunun için argümanları göreceğiz.
 *mesela 1.den 2.ye veri yollayacağız diyelim 2. fragmente basıyoruz my_graph'da
 ve argümanlar kısmında +'ya basıyoruz age adında integer bir argüman tanımlıyoruz.
 sonra tekrar build rebuild yapıyoruz.!!!


 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }
}

File: KotlinDers12Navigation/app/src/main/java/com/example/kotlinders12navigation/MainActivity.kt
package com.example.kotlinders12navigation

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

/**
 * Jetpack: Geliştiriciler için oluşturulmuş bazı kütüphaneler
 * Fragment de bir jetpack
 * uygulama içersindeki navigasyonu sağlıyor. Özelllikle fragmentlerde kullanılıyor
 *developer.android.com a git -jetpack-navigation-ordan dependencies kısımları için eklemelerini al(Kotlin için olanları)
 *Navigation kullanımı:
 -res-new-android resource file-resource type=Navigation seç!!-navigation_graph isimli bir xml doyası oluşturduk
fragmentlerimi o xml dosyasında istediğim gibi yerleştirebiliyorum
 -fragmente tıklayınca yan tarafta attributesler çıkıyor.
 -fragment 1 den 2 ye bir ok çıkardık bu aksiyon oluyor. Nereye gideceğini belirttik.
-MainActivity.xml dosyama navHostFragment ekledim ordan oluşturduğum navigation_graph isimli navigation'umu seçtim
 -birbirleri arasındaki geçişi fragment.kt sınıflarında yapıyoruz action oluşturarak.
 *Argümanlar:
 -fragmentler arasında veri göndermek istiyoruz. Mesela birinci fragmentden 2.ye bir veri aktarmak
 istiyoruz diyelim 2.fragmente bir argüman oluşturmamız gerekiyor.
 2.fragmente bas argüman ekle sanki bir değişken ekliyormuş gibi
 *
 *
 *
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }
}
